### PR TITLE
xautocfg: init at 1.2

### PR DESCRIPTION
**Pull request by jceb**
## Description of changes

Add xautocfg, a tool to automatically configure the repeat rate for new keyboards.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested, as applicable:
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  libX11,
+  libXi,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xautocfg";
+  version = "1.2";
+  src = fetchFromGitHub {
+    owner = "SFTtech";
+    repo = pname;
```
Comment (Sigmanificient) [inline pkgs/by-name/xa/xautocfg/package.nix:14]: - #277994
```suggestion
    repo = "xautocfg";
```

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  stdenv,
+  fetchFromGitHub,
+  libX11,
+  libXi,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xautocfg";
+  version = "1.2";
+  src = fetchFromGitHub {
```
Comment (Sigmanificient) [inline pkgs/by-name/xa/xautocfg/package.nix:12]:
```suggestion
  version = "1.2";

  src = fetchFromGitHub {
```

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  libX11,
+  libXi,
+}:
+
+stdenv.mkDerivation rec {
```
Comment (Sigmanificient) [inline pkgs/by-name/xa/xautocfg/package.nix:9]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "xautocfg";
+  version = "1.2";
+  src = fetchFromGitHub {
+    owner = "SFTtech";
+    repo = pname;
+    rev = "v${version}";
```
Comment (Sigmanificient) [inline pkgs/by-name/xa/xautocfg/package.nix:15]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "MANPREFIX=$(out)"
+  ];
+
+  meta = {
+    homepage = "https://github.com/SFTtech/xautocfg";
+    description = "Automatic keyboard repeat rate configuration for new keyboards";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ jceb ];
+  };
+}
```
Comment (Sigmanificient) [inline pkgs/by-name/xa/xautocfg/package.nix:35]:
```suggestion
})
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  makeFlags = [
+    "PREFIX=$(out)"
+    "MANPREFIX=$(out)"
+  ];
+
+  meta = {
+    homepage = "https://github.com/SFTtech/xautocfg";
+    description = "Automatic keyboard repeat rate configuration for new keyboards";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ jceb ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/xa/xautocfg/package.nix:34]:
```suggestion
    maintainers = with lib.maintainers; [ jceb ];
    mainProgram = "xautocfg";
```

**Comment by jceb** [comment]
Is there anything else missing in this PR before it can be merged?

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  buildInputs = [
+    libX11
+    libXi
+  ];
+
+  makeFlags = [
+    "PREFIX=$(out)"
+    "MANPREFIX=$(out)"
```
Comment (Aleksanaa) [inline pkgs/by-name/xa/xautocfg/package.nix:27]:
```suggestion
    "PREFIX=${placeholder "out"}"
    "MANPREFIX=${placeholder "out"}"
```

**Comment by Aleksanaa** [comment]
@ofborg build xautocfg

File: pkgs/by-name/xa/xautocfg/package.nix [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    "PREFIX=${placeholder "out"}"
+    "MANPREFIX=${placeholder "out"}"
+  ];
+
+  meta = {
+    homepage = "https://github.com/SFTtech/xautocfg";
+    description = "Automatic keyboard repeat rate configuration for new keyboards";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ jceb ];
+    mainProgram = "xautocfg";
```
Comment (Aleksanaa) [inline pkgs/by-name/xa/xautocfg/package.nix:35]:
```suggestion
    mainProgram = "xautocfg";
    platforms = lib.platforms.linux;
```

**Comment by Aleksanaa** [comment]
@ofborg build xautocfg

**Pull request merged**